sendgrid/sendgrid.py

```diff
@@ -109,6 +113,41 @@ def _make_request(self, message):
         body = response.read()
         return response.getcode(), body
 
+    def _make_persistent_request(self, message):
+        data = urlencode(self._build_body(message), True).encode('utf-8')
+        domain = self.host.lstrip('http://').lstrip('https://')
+        headers = {
+            'User-Agent': self.useragent,
+            'Host': domain + ':' + self.port,
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Connection': 'Keep-Alive',
+        }
+        if self.username is None:
+            # Using API key
+            headers['Authorization'] = 'Bearer ' + self.password
+        for _ in range(self._max_retry):
+            if self._server is None:
+                self._server = http_client.HTTPSConnection(domain,
+                                                           int(self.port))
+            self._server.request('POST', self.endpoint, data, headers=headers)
+            try:
+                response = self._server.getresponse()
+                body = response.read()
+                return response.status, body
+            except http_client.BadStatusLine:
+                # Persistent request timeout reached. Retrying...
+                self._server.close()
+                self._server = None
+        raise timeout("Unable to start persistent connection")
+
+    def _make_request(self, message):
+        if not self.proxies:
+            try:
+                return self._make_persistent_request(message)
+            except timeout:

```

Comment:
Perhaps the comment could be `Timeout on persistent request, falling back to non-persistent request`.

Btw, will falling back to non-persistent requests going to help?